executorCommon/src/main/scala/dx/executor/JobMeta.scala

```diff
@@ -403,6 +403,35 @@ abstract class JobMeta(val workerPaths: DxWorkerPaths,
     }
   }
 
+  def extendFileDescCache(newFiles: Vector[DxFile]): DxFileDescCache = {
+    DxFileDescCache(
+        allFilesReferenced ++ safeBulkDescribe(queryFiles = newFiles, verifyClosed = false)
+    )
+  }
+
+  private def safeBulkDescribe(queryFiles: Vector[DxFile],
+                               verifyClosed: Boolean = true): Vector[DxFile] = {
+    logger.trace(s"Bulk describing ${queryFiles.size} files")
+    val dxFiles = dxApi.describeFilesBulk(queryFiles, searchWorkspaceFirst = true, validate = true)
+    // check that all files are in the closed state
+
+    val notClosed = {
+      if (verifyClosed) {
+        logger.trace(s"Checking that all files are closed")
+        dxFiles.filterNot(_.describe().state == DxState.Closed)
+      } else {
+        Vector.empty
+      }
+    }
+    if (notClosed.nonEmpty) {
```

Comment:
If the files are not closed (e.g in the process of closing) would it make sense to retry or wait before throwing an error? Or we expect a non-closed file to remain open/closing at this point anyway?

Reply:
dxCompiler `-waitOnUpload` covers that. Previous behavior was that if `-waitOnUpload` is not active - no checks were done to the delocalized files at all, and `file-xxx/described` was called only to get a URI of the file on `dx://` I basically preserve this behavior.

Reply:
OK, sg, thanks!